source/main.cpp

```diff
@@ -0,0 +1,435 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2021 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cctype>
+#include <chrono>
+#include <events/mbed_events.h>
+
+#include "ble/BLE.h"
+#include "pretty_printer.h"
+
+/** This program enters different BLE states according to operator input, allowing power consumption to be
+ * measured.
+ */
+
+using namespace std::literals::chrono_literals;
+
+static const char DEVICE_NAME[] = "Power Consumption";
+static const uint16_t MAX_ADVERTISING_PAYLOAD_SIZE = 50;
+
+static const ble::scan_duration_t SCAN_TIME(1000);
+static const events::EventQueue::duration CONNECT_TIME(10000);
+
+static const ble::adv_duration_t ADVERTISE_TIME(1000);
```

Comment:
It would be worth increasing these values otherwise there won't be much radio activity.